### PR TITLE
build: update supported python versions to include 3.12

### LINKED ISSUE #38: Update supported python versions
**Is your feature request related to a problem? Please describe.**
[Here](https://github.com/biocommons/biocommons.github.io/blob/c1ac95dcf2b290ddd546af8e22132b1c0297de51/docs/contributing/coding-guidelines.md#other) the biocommons site says:
>We support most recent 3 Python releases; https://devguide.python.org/versions/
>
>     3.9, 3.10, 3.11 now
>     drop 3.9 and add 3.12 when 3.12 is available

**Describe the solution you'd like**
The following should be changed to use python 3.10, 3.11, and 3.12 since 3.12 is now a [supported version](https://devguide.python.org/versions/#supported-versions):
https://github.com/biocommons/biocommons.example/blob/4b0e4bf44788f30b385eeef32dd5a63410f47b44/pyproject.toml#L12-L14
https://github.com/biocommons/biocommons.example/blob/4b0e4bf44788f30b385eeef32dd5a63410f47b44/.github/workflows/python-package.yml#L53

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by korikuzma**
close #38

* support the 3 most recent python releases: 3.10, 3.11, 3.12

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [tox]
-envlist = py39,py310
+envlist = py310,py311,py312
 isolated_build = True
 
 [gh]
 python =
-    3.9 = py39, type
```
Comment (korikuzma) [inline tox.ini:7]: not sure what the type was for. Do I need to add this back?

**Review by jsstevenson — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (jsstevenson) [reply tox.ini:7]: IMO delete it here and discuss in #43

**Comment by korikuzma** [comment]
@reece how can we update all repos with this change?

**Comment by reece** [comment]
At maintainers meeting, agreed to upgrade repos to importlib (and remove pkg_resources) before supporting 3.12.

**Comment by jsstevenson** [comment]
I've created test PRs for most of the major repos (hesitant to do too much in UTA since I know it's undergoing a significant refactor). Some are showing errors in 3.12 that should be dealt with before this goes through.

**Review by jsstevenson — APPROVED** [review]
🚀

**Review by reece — APPROVED** [review]

**Pull request merged**